src/newTest/java/org/opensearch/test/framework/cluster/TestRestClient.java

```diff
@@ -0,0 +1,366 @@
+    }
+
+    public HttpResponse getAuthInfo( Header... headers) throws Exception {
+        return executeRequest(new HttpGet(getHttpServerUri() + "/_opendistro/_security/authinfo?pretty"), headers);
+    }
+
+    public HttpResponse head(String path, Header... headers) throws Exception {
+        return executeRequest(new HttpHead(getHttpServerUri() + "/" + path), headers);
+    }
+
+    public HttpResponse options(String path, Header... headers) throws Exception {
+        return executeRequest(new HttpOptions(getHttpServerUri() + "/" + path), headers);
+    }
+
+    public HttpResponse putJson(String path, String body, Header... headers) throws Exception {
+        HttpPut uriRequest = new HttpPut(getHttpServerUri() + "/" + path);
+        uriRequest.setEntity(new StringEntity(body));
+
+        HttpResponse response = executeRequest(uriRequest, mergeHeaders(CONTENT_TYPE_JSON, headers));
+
+        if (response.getStatusCode() < 400 && trackResources && !puttedResourcesSet.contains(path)) {
+            puttedResourcesSet.add(path);
+            puttedResourcesList.add(path);
+        }
+
+        return response;
+    }
+
+    public HttpResponse putJson(String path, ToXContentObject body) throws Exception {
+        return putJson(path, Strings.toString(body));
+    }
+
+    public HttpResponse put(String path) throws Exception {
+        HttpPut uriRequest = new HttpPut(getHttpServerUri() + "/" + path);
+        HttpResponse response = executeRequest(uriRequest);
+
+        if (response.getStatusCode() < 400 && trackResources && !puttedResourcesSet.contains(path)) {
+            puttedResourcesSet.add(path);
+            puttedResourcesList.add(path);
+        }
+
+        return response;
+    }
+
+    public HttpResponse delete(String path, Header... headers) throws Exception {
+        return executeRequest(new HttpDelete(getHttpServerUri() + "/" + path), headers);
+    }
+
+    public HttpResponse postJson(String path, String body, Header... headers) throws Exception {
+        HttpPost uriRequest = new HttpPost(getHttpServerUri() + "/" + path);
+        uriRequest.setEntity(new StringEntity(body));
+        return executeRequest(uriRequest, mergeHeaders(CONTENT_TYPE_JSON, headers));
+    }
+
+    public HttpResponse postJson(String path, ToXContentObject body) throws Exception {
+        return postJson(path, Strings.toString(body));
+    }
+
+    public HttpResponse post(String path) throws Exception {
+        HttpPost uriRequest = new HttpPost(getHttpServerUri() + "/" + path);
+        return executeRequest(uriRequest);
+    }
+
+    public HttpResponse patch(String path, String body) throws Exception {
+        HttpPatch uriRequest = new HttpPatch(getHttpServerUri() + "/" + path);
+        uriRequest.setEntity(new StringEntity(body));
+        return executeRequest(uriRequest, CONTENT_TYPE_JSON);
+    }
+
+    public HttpResponse executeRequest(HttpUriRequest uriRequest, Header... requestSpecificHeaders) throws Exception {
```

Comment:
Catch, wrap and throw runtime exception, avoid hanlding checked exceptions in the tests

Reply:
done